airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java

```diff
@@ -63,23 +64,74 @@ public class TemporalClient {
    */
   private static final int DELAY_BETWEEN_QUERY_MS = 10;
 
-  private static final int MAXIMUM_SEARCH_PAGE_SIZE = 50;
+  public static TemporalClient production(final Configs configs) {
+    if (configs.temporalCloudEnabled()) {
+      log.info("TemporalClient.production chose Cloud...");
+      return TemporalClient.cloud(configs);
+    }
+    log.info("TemporalClient.production chose Airbyte...");
+    return TemporalClient.airbyte(configs);
+  }
+
+  public static WorkerFactory productionWorkerFactory(final Configs configs) {
+    if (configs.temporalCloudEnabled()) {
+      log.info("TemporalClient.productionWorkerFactory chose Cloud...");
+      return WorkerFactory.newInstance(cloudWorkflowClient(TemporalUtils.createTemporalCloudService(), configs));
+    }
+    log.info("TemporalClient.productionWorkerFactory chose Airbyte...");
+    return WorkerFactory.newInstance(airbyteWorkflowClient(TemporalUtils.createTemporalAirbyteService()));
+  }
 
-  public static TemporalClient production(final String temporalHost, final Path workspaceRoot, final Configs configs) {
-    final WorkflowServiceStubs temporalService = TemporalUtils.createTemporalService(temporalHost);
-    return new TemporalClient(WorkflowClient.newInstance(temporalService), workspaceRoot, temporalService, configs);
+  public static TemporalClient cloud(final Configs configs) {
```

Comment:
If an external caller needs to know if its talking to cloud or to airbyte-temporal, these methods are available.

The migrator app is the only current use case, and after the migration is finished, we can consider making these private

Reply:
got it. I felt the various create methods are confusing. what about consolidating all these into a top level `TemporalClient.get` method all callers use? in this manner, we will not need to come back and remove code.

Reply:
I think the existing `TemporalClient.production` method is essentially what you're describing with a top level `TemporalClient.get` method.

But, the migrator is unique in that it needs to explicitly instantiate a cloud client, and also an airbyte client. So it can't call `TemporalClient.production`, it has to make two separate calls to get two separate clients. Does that follow? Want to make sure we're understanding eachother

Reply:
Good call! Added.